app/controllers/contributions_controller.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+class ContributionsController < ApplicationController
+  load_and_authorize_resource class: TaskContribution, except: %i[create new]
+  def approve_changes
+    contrib = TaskContribution.find(params[:contribution_id])
+    @task = Task.find(params[:task_id])
+    if @task.apply_contribution(contrib)
```

Comment:
Something like this could work (adopted for the given scenario):

```
raise Pundit::NotAuthorizedError if @uef.present? && @uef.exercise != @exercise
```

Reply:
I was still able to manipulate the URL

`http://localhost:7500/tasks/23/task_contributions/2` is invalid, since the contribution is for task `24`. However, this is not checked correctly.

Reply:
I'm sorry. I thought this scenario was already covered by a different test, but it seems like I was wrong.
The correct fix is now implemented.

Reply:
Keeping this thread open. The following are left:

1. Migrating the remaining mailers to Slim files (could be another PR).
2. Dropping the outer HTML parts of the reaming mailers (could be another PR).
3. Dropping outdated mailers and views (i.e., `send_contribution_request`) which are now superseded by your changes.
4. Moving hard-coded strings to locales
5. Either using the `current_user.preferred_locale` or adding both languages to a mail.
6. In text emails, do not use any HTML tags (i.e., no `<br>`, rather `== "\n\n"`). See the other text mailers.